x-pack/legacy/plugins/siem/public/components/timeline/insert_timeline_popover/popover.tsx

```diff
@@ -0,0 +1,302 @@
+            </MyEuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiIcon
+            type={`${
+              option.favorite != null && isEmpty(option.favorite) ? 'starEmpty' : 'starFilled'
+            }`}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }, []);
+
+  const handleTimelineChange = useCallback(
+    options => {
+      const selectedTimeline = options.filter(
+        (option: { checked: string }) => option.checked === 'on'
+      );
+      if (selectedTimeline != null && selectedTimeline.length > 0) {
+        onTimelineChange(
+          isEmpty(selectedTimeline[0].title)
+            ? i18nTimeline.UNTITLED_TIMELINE
+            : selectedTimeline[0].title,
+          selectedTimeline[0].id
+        );
+      }
+      setIsPopoverOpen(false);
+    },
+    [onTimelineChange]
+  );
+
+  const handleOnScroll = useCallback(
+    (
+      totalTimelines: number,
+      totalCount: number,
+      {
+        clientHeight,
+        scrollHeight,
+        scrollTop,
+      }: {
+        clientHeight: number;
+        scrollHeight: number;
+        scrollTop: number;
+      }
+    ) => {
+      if (totalTimelines < totalCount) {
+        const clientHeightTrigger = clientHeight * 1.2;
+        if (
+          scrollTop > 10 &&
+          scrollHeight - scrollTop < clientHeightTrigger &&
+          scrollHeight > heightTrigger
+        ) {
+          setHeightTrigger(scrollHeight);
+          setPageSize(pageSize + ORIGINAL_PAGE_SIZE);
+        }
+      }
+    },
+    [heightTrigger, pageSize]
+  );
+
+  const insertTimelineButton = useMemo(
+    () => (
+      <EuiButtonIcon
+        aria-label={i18n.INSERT_TIMELINE}
+        data-test-subj="insert-timeline-button"
+        iconType="timeline"
+        isDisabled={isDisabled}
+        onClick={handleOpenPopover}
+      />
+    ),
+    [handleOpenPopover, isDisabled]
+  );
+
+  const favoritePortal = useMemo(
+    () =>
+      searchRef != null ? (
+        <EuiPortal insert={{ sibling: searchRef, position: 'after' }}>
+          <MyEuiFlexGroup gutterSize="xs" justifyContent="flexEnd">
+            <EuiFlexItem grow={false}>
+              <EuiFilterGroup>
+                <EuiFilterButton
+                  size="l"
+                  data-test-subj="only-favorites-toggle"
+                  hasActiveFilters={onlyFavorites}
+                  onClick={handleOnToggleOnlyFavorites}
+                >
+                  {i18nTimeline.ONLY_FAVORITES}
+                </EuiFilterButton>
+              </EuiFilterGroup>
+            </EuiFlexItem>
+          </MyEuiFlexGroup>
+        </EuiPortal>
+      ) : null,
+    [searchRef, onlyFavorites, handleOnToggleOnlyFavorites]
+  );
+
+  return (
+    <EuiPopover
+      id="searchTimelinePopover"
+      button={insertTimelineButton}
+      isOpen={isPopoverOpen}
+      closePopover={handleClosePopover}
+    >
+      <AllTimelinesQuery
```

Comment:
I think some of this code looks like that it can be common between this component and search_super_select. what do you think?

Reply:
yep, you'd prefer i just combine them?

Reply:
i was on the edge of whether or not to keep it as one or 2. there was a bit of difference because the super select auto selects, and we don't show the default timeline as one you can insert. check the diff and let me know what you think: https://www.diffchecker.com/OVB7s83l

Reply:
I like the two components but we should have a generic component to manage this 
![image](https://user-images.githubusercontent.com/189600/76275313-5b347f80-6259-11ea-9bc2-0332a99c8ec8.png)

Reply:
check it out now plz

Reply:
I was more thinking to create a component who manages the query of the timelines and keeps the two components that you had before it is more flexible and should be easier to test. I did push my idea of refactor in your PR, if you disagree let's discuss it.

https://github.com/elastic/kibana/pull/59586/commits/5f38bc41300e4f8890c3a69023f46191b9acdd8a

Reply:
wow thats great @XavierM , thanks you da man 🎸